### DISCUSSION TITLE
`mesa` doesn't install `libEGL` even though it's set to do so.

### DISCUSSION THREAD

**Discussion by jmuchovej** [Homebrew/discussions#3766] [Tap maintenance and Homebrew development]
👋 I'm trying to install `EGL` for a project I'm working on and looking through the `mesa` Formula, I was hopeful. 😅

Based on https://github.com/Homebrew/homebrew-core/blob/c3a39e90949924d976778ee57ffcc2ddb4016f9e/Formula/mesa.rb#L111, I believe `libEGL.dylib` should be installed at `$(brew --cellar)/mesa/<version>/lib/`, but this is what I end up with:

```shell
$ ls -l $(brew --cellar)/mesa/22.2.0/lib                         
total 964K
drwxr-xr-x 3 jm staff   96 Oct  9 21:44 dri/
drwxr-xr-x 6 jm staff  192 Oct  9 21:44 pkgconfig/
lrwxr-xr-x 1 jm staff   13 Oct  9 21:44 libGL.dylib -> libGL.1.dylib
lrwxr-xr-x 1 jm staff   20 Oct  9 21:44 libGLESv1_CM.dylib -> libGLESv1_CM.1.dylib
-r--r--r-- 1 jm staff  58K Oct  9 21:44 libGLESv1_CM.1.dylib
lrwxr-xr-x 1 jm staff   17 Oct  9 21:44 libGLESv2.dylib -> libGLESv2.2.dylib
-r--r--r-- 1 jm staff  86K Oct  9 21:44 libGLESv2.2.dylib
-r--r--r-- 1 jm staff 443K Oct  9 21:44 libGL.1.dylib
lrwxr-xr-x 1 jm staff   16 Oct  9 21:44 libglapi.dylib -> libglapi.0.dylib
-r--r--r-- 1 jm staff 369K Oct  9 21:44 libglapi.0.dylib
```

I'm not sure if this is a bug or if I've misunderstood the Mesa docs, so some guidance here would be much appreciated. 🙂 (`EGL` is the primary lib that I need and I can't get around it – it's required by one of the core libraries I'm hoping to use.)

Edit: LOL. I completely missed the `if OS.linux?` line. So it seems that `-D egl=true` isn't passed for MacOS. 😔

**Comment by arrmo** [comment]
Seeing a similar issue here - sorry, are you saying you figured out a fix?

Thanks!

**Reply by jmuchovej** [reply]
I ended up taking a different approach to that project, so I never found a fix, no. 😕 

Though I believe the `mesa` Formula could be updated to include some of the options from `if OS.linux?` and it would compile – but it's been a few months since I tried it and don't have the logs (to check if it worked or not).

**Reply by arrmo** [reply]
NP, thanks! I admit, having issues getting CMake to see the Homebrew libraries. Hmmm.

Thanks!
